### PR TITLE
Migrated integration tests

### PR DESCRIPTION
**Pull request by heiko-braun**

### CODE REVIEW

**Comment by heiko-braun** [comment]
@Ladicek Here's the PR I've been talking about

File: app/pom.xml [hunk 1]
```diff
@@ -71,6 +84,13 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
+      <groupId>org.keycloak</groupId>
+      <artifactId>keycloak-authz-client</artifactId>
+      <version>2.5.7.Final-redhat-2</version>
```
Comment (Ladicek) [inline app/pom.xml]: Maybe not a good idea for the community variant of the booster?

File: app/src/test/java/io/openshift/booster/SsoIT.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        AccessTokenResponse accessTokenResponse = authzClient.obtainAccessToken("admin", "admin");
+        for (String url : applicationUrls) {
+            try {
+                getGreeting(url, accessTokenResponse.getToken(), null);
+                fail("ForbiddenException expected");
+            } catch (ForbiddenException e) {
+                // expected
+            }
+        }
+    }*/
```
Comment (Ladicek) [inline app/src/test/java/io/openshift/booster/SsoIT.java]: I thought this test passes now?

File: app/pom.xml (hunk 1 above)
Comment (heiko-braun) [reply app/pom.xml]: Don't know. There is another request to baseline on this version. Let's take it offline

File: app/src/test/java/io/openshift/booster/SsoIT.java [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+    @Test
+    public void badPassword() {
+        try {
+            AccessTokenResponse accessTokenResponse = authzClient.obtainAccessToken("alice", "bad");
+            fail("HttpResponseException expected (401)");
+        } catch (Throwable t) {
```
Comment (Ladicek) [inline app/src/test/java/io/openshift/booster/SsoIT.java:140]: Should check if it's the correct exception.

File: app/src/test/java/io/openshift/booster/SsoIT.java (hunk 2 above)
Comment (heiko-braun) [reply app/src/test/java/io/openshift/booster/SsoIT.java]: I'll have to double check. But to me the assertion is unclear. Why would an admin user not be allowed to request a token?

File: app/pom.xml (hunk 1 above)
Comment (kenfinnigan) [reply app/pom.xml]: Should be 2.5.5.Final for community

File: pom.xml [hunk 4]
```diff
@@ -50,4 +50,50 @@
...
+      <releases>
+        <enabled>true</enabled>
+      </releases>
+      <snapshots>
+        <enabled>false</enabled>
+      </snapshots>
+      <id>jboss-enterprise-maven-repository-earlyaccess</id>
+      <url>https://maven.repository.redhat.com/earlyaccess/all/</url>
+    </pluginRepository>
+  </pluginRepositories>
```
Comment (Ladicek) [inline pom.xml]: This is all inherited from booster-parent.

File: app/src/test/java/io/openshift/booster/SsoIT.java (hunk 2 above)
Comment (Ladicek) [reply app/src/test/java/io/openshift/booster/SsoIT.java]: It's no longer `@Ignore`d in the SSO submodule.

And indeed the `admin` user is allowed to request a token, but doesn't have the correct role. This test checks the "authenticated, but not authorized" flow.

**Comment by Ladicek** [comment]
Personally, I'd kill JAX-RS Client with fire and just used Apache HTTP Client. Disabling hostname verification and certificate check is as easy as https://gist.github.com/Ladicek/fe783b7f871e33f5cf4c403065714ddd

File: app/src/test/java/io/openshift/booster/SsoIT.java (hunk 2 above)
Comment (heiko-braun) [reply app/src/test/java/io/openshift/booster/SsoIT.java]: I understand. Thanks for the clarification.

**Comment by heiko-braun** [comment]
I'll leave the JAX-RS client for one reason: It's the Swarm way to most things HTTP. And I believe it's useful to show people how to work with token use cases in a Swarm idiomatic way.

**Comment by Ladicek** [comment]
I don't necessarily agree that JAX-RS Client is Swarm-idiomatic, but whatever. (Actually the way we generate a client from an annotated interface isn't JAX-RS at all, just a proprietary extension...)

File: app/src/test/java/io/openshift/booster/SsoIT.java [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+    }
+
+    @Test
+    public void badPassword() {
+        try {
+            AccessTokenResponse accessTokenResponse = authzClient.obtainAccessToken("alice", "bad");
+            fail("HttpResponseException expected (401)");
+        } catch (Throwable t) {
+            assertThat(t).isInstanceOf(HttpResponseException.class);
+            assertThat(((HttpResponseException)t).getStatusCode()).isEqualTo(401);
```
Comment (Ladicek) [inline app/src/test/java/io/openshift/booster/SsoIT.java:142]: I think it should be `NotAuthorizedException` -- could check that similarly to `adminUser`. But this is good too.

File: app/src/test/java/io/openshift/booster/SsoIT.java [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+            Greeting greeting = getGreeting(url, accessTokenResponse.getToken(), "Scott");
+
+            assertThat(greeting).isNotNull();
+            assertThat(greeting.getContent()).contains("Hello, Scott!");
+        }
+    }
+
+    // This test checks the "authenticated, but not authorized" flow.
+    @Test
+    @Ignore
```
Comment (Ladicek) [inline app/src/test/java/io/openshift/booster/SsoIT.java]: Does it fail currently?

Comment (heiko-braun) [reply app/src/test/java/io/openshift/booster/SsoIT.java]: yes, it does. see my comments on irc

**Comment by heiko-braun** [comment]
@Ladicek I think with the latest commit we are good to go.

File: app/src/main/java/io/openshift/booster/GreetingController.java [hunk 7]
```diff
@@ -17,6 +17,7 @@
 
 import java.util.concurrent.atomic.AtomicLong;
 
+import javax.annotation.security.RolesAllowed;
```
Comment (Ladicek) [inline app/src/main/java/io/openshift/booster/GreetingController.java:20]: Useless.

File: app/src/test/resources/keycloak.json [hunk 8]
```diff
@@ -0,0 +1,8 @@
+{
+  "realm": "master",
+  "auth-server-url" : "${sso.auth.server.url}",
+  "resource" : "demoapp",
```
Comment (Ladicek) [inline app/src/test/resources/keycloak.json:4]: Still using `demoapp`?

Comment (heiko-braun) [reply app/src/test/resources/keycloak.json:4]: That's used to retrieve the token. According to Paulo there are two clients configured: `demoapp` for managing tokens and `secure-booster-endpoint` for accessing the greeting service. Take a look at the sso booster doc capturing the docs changes

Comment (Ladicek) [reply app/src/test/resources/keycloak.json:4]: Aha, OK. I don't pretend I understand it, but whatever.

**Review by Ladicek — APPROVED** [review]

**Pull request merged**